app/github/push_event.rb

```diff
@@ -41,9 +45,9 @@ def deleted?
     event['deleted']
   end
 
-  # The git commit sha to deploy
+  # The git commit sha of this push event.
   def sha
-    event['head_commit']['id']
+    event['head_commit']['id'] unless event['head_commit'].nil?
```

Comment:
No need for `.nil?`, you can just do:

```ruby
event['head_commit']['id'] unless event['head_commit']
```

Reply:
@ejholmes when I try this many tests fail.

Reply:
uugh, Rubies `unless` hurts my head.  : )

I basically wanted Python's 

```
return event['head_commit']['id'] if event['head_commit'] is not None
```

which is how I sort of explicitly wrote it.

The implicit (and more inclusive) Python form would be 

```
return event['head_commit']['id'] if event['head_commit']
```

This translates to Ruby as simply 

```  # The git commit sha of this push event.
  def sha
    event['head_commit']['id'] if event['head_commit']
  end```

Reply:
Woops, sorry. Made a typo in my comment. Meant:

```diff
- event['head_commit']['id'] unless event['head_commit']
+ event['head_commit']['id'] if event['head_commit']
```